### PR TITLE
Update archive feature

### PR DESCRIPTION
**Pull request by cretadn22**
### Fixed Issues

$ https://github.com/Expensify/App/issues/45881
PROPOSAL: https://github.com/Expensify/App/issues/45881#issuecomment-2254130299

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Tap fab - create a new workspace
2. Tap on workspace chat
3. Submit a expense using plus icon near compose
4. Navite to LHN and note green dot next to workspace chat
5. Tap profile icon -- workspaces -- workspace
6. Tap 3 dots next to workspace and delete the workspace
7. Navigate to LHN
8. Verify that archived workspace must not show green dot in LHN
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/37da68f2-12e1-4484-8b82-ef3909a99c53

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/050cdc2a-7e49-4199-a958-601f8e0926d3

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/cb1ff88d-b51d-4fb0-a210-b04746012f80

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/c9bd2bdf-26e6-4889-98ce-fc7d5358b70e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/25401eaa-08d5-487b-8203-6ab4d9cb6762

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/a3894929-6952-4710-8897-afe789eff403

</details>

### CODE REVIEW

**Review by situchan — COMMENTED** [review]
are you sure that there's no more `isArchivedRoom` occurrences to be replaced with `private_isArchived`?

**Comment by situchan** [comment]
> I reverted the isArchivedRoom to prevent any regression. I believe it's preferable to use it this way rather than calling the isArchivedRoom function individually for each case.

It's against what @srikarparsi said [here](https://github.com/Expensify/App/issues/45881#issuecomment-2270414860)

**Comment by situchan** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/libs/SidebarUtils.ts [hunk 1]
```diff
@@ -278,7 +278,8 @@ function getOptionData({
         hasParentAccess: undefined,
         isIOUReportOwner: null,
         isChatRoom: false,
-        isArchivedRoom: false,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        private_isArchived: undefined,
```
Comment (situchan) [inline src/libs/SidebarUtils.ts:282]: This `result` object type is `ReportUtils.OptionData`, but `ReportUtils.OptionData` still has element `isArchivedRoom?: boolean | null;`, not `private_isArchived`.

File: src/libs/ReportUtils.ts [hunk 2]
```diff
@@ -428,6 +428,7 @@ type OptionData = {
     hasDraftComment?: boolean | null;
     keyForList?: string;
     searchText?: string;
+    private_isArchived?: string | null;
     isIOUReportOwner?: boolean | null;
     isArchivedRoom?: boolean | null;
```
Comment (situchan) [inline src/libs/ReportUtils.ts:433]: @srikarparsi is it fine to keep both (private_isArchived, isArchivedRoom) in OptionData?

Comment (srikarparsi) [reply src/libs/ReportUtils.ts:434]: Sorry, didn't see this. Please correct me if I'm wrong but OptionData includes all the attributes from Report right (which already has private_isArchived)? So do we need to declare this twice?

**Comment by srikarparsi** [comment]
Hi @situchan, do you think you could take another look at this when you have a chance?

**Comment by situchan** [comment]
(Not related but worth mentioning)

App crashes after deleting workspace from workspaces list:

crash

**Review by situchan — APPROVED** [review]
Looks good.
Hope we haven't missed anything!

File: src/libs/OptionsListUtils.ts [hunk 3]
```diff
@@ -2301,7 +2301,7 @@ function getHeaderMessageForNonUserList(hasSelectableOptions: boolean, searchVal
  * Helper method to check whether an option can show tooltip or not
  */
 function shouldOptionShowTooltip(option: ReportUtils.OptionData): boolean {
-    return (!option.isChatRoom || !!option.isThread) && !option.isArchivedRoom;
+    return (!option.isChatRoom || !!option.isThread) && !ReportUtils.isArchivedRoom(option, ReportUtils.getReportNameValuePairs(option?.reportID));
```
Comment (srikarparsi) [inline src/libs/OptionsListUtils.ts:2304]: Can we change [this](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts#L428) from isArchivedRoom to private_isArchived. And then we can directly use option.private_isArchived.

**Review by srikarparsi — CHANGES_REQUESTED** [review]

File: src/libs/OptionsListUtils.ts (hunk 3 above)
Comment (srikarparsi) [reply src/libs/OptionsListUtils.ts:2304]: Hm, I think we're actually trying to work towards standardizing using private_isArchived. I think the entire point of the option is to have data about the report. The Report data type already has private_isArchived so all we would have to do is remove this [line](https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts#L428) which unnecessarily declares a different variable and then use the current variable (private_isArchived) which has the right data.

Comment (srikarparsi) [reply src/libs/OptionsListUtils.ts:2304]: We're in the process of moving some keys from the `reports` collection to a new collection named `reportNameValuePairs` which is why this is currently in the codebase. When we do that, we will most likely need to create a new `reportNameValuePairs` data type similar to the reports data type and move all related keys (along with the private_isArchived key that we will change to using in this PR).

File: src/libs/OptionsListUtils.ts [hunk 4]
```diff
@@ -1588,7 +1587,8 @@ function orderOptions(options: ReportUtils.OptionData[], searchValue: string | u
                 if (preferChatroomsOverThreads && option.isThread) {
                     return 4;
                 }
-                if (!!option.isChatRoom || option.isArchivedRoom) {
+                const reportNameValuePairs = ReportUtils.getReportNameValuePairs(option?.reportID);
+                if (!!option.isChatRoom || ReportUtils.isArchivedRoom(option, reportNameValuePairs)) {
```
Comment (srikarparsi) [inline src/libs/OptionsListUtils.ts:1591]: Can we use `option.private_isArchived` here as well? OptionData extends the Report data type which has `private_isArchived`

File: src/libs/OptionsListUtils.ts [hunk 5]
```diff
@@ -1833,7 +1833,8 @@ function getOptions(
     // - All archived reports should remain at the bottom
     const orderedReportOptions = lodashSortBy(filteredReportOptions, (option) => {
         const report = option.item;
-        if (option.isArchivedRoom) {
+        const reportNameValuePairs = ReportUtils.getReportNameValuePairs(option?.reportID);
+        if (ReportUtils.isArchivedRoom(option, reportNameValuePairs)) {
```
Comment (srikarparsi) [inline src/libs/OptionsListUtils.ts:1837]: Same thing here I believe?

File: tests/unit/OptionsListUtilsTest.ts [hunk 6]
```diff
@@ -2639,9 +2639,11 @@ describe('OptionsListUtils', () => {
             const searchText = 'Archived';
             const options = OptionsListUtils.getSearchOptions(OPTIONS, '', [CONST.BETAS.ALL]);
             const filteredOptions = OptionsListUtils.filterOptions(options, searchText);
+            const reportNameValuePairs = ReportUtils.getReportNameValuePairs(filteredOptions.recentReports[0]?.reportID);
+            const isArchivedRoom = ReportUtils.isArchivedRoom(filteredOptions.recentReports[0], reportNameValuePairs);
 
             expect(filteredOptions.recentReports.length).toBe(1);
-            expect(filteredOptions.recentReports[0].isArchivedRoom).toBe(true);
```
Comment (srikarparsi) [inline tests/unit/OptionsListUtilsTest.ts:2644]: This test should be `filteredOptions.recentReports[0].private_isArchived`

**Review by srikarparsi — CHANGES_REQUESTED** [review]
Sorry, I'm not sure if I wasn't clear but since OptionData has private_isArchived (since it extends report), we should just be using that field instead of getting the reportNameValuePairs and then passing it into isArchivedRoom

File: src/libs/OptionsListUtils.ts (hunk 5 above)
Comment (srikarparsi) [reply src/libs/OptionsListUtils.ts:1837]: There's a couple other places where the same thing occurs. I left a comment [here](https://github.com/Expensify/App/pull/46934#pullrequestreview-2301889226)

**Comment by srikarparsi** [comment]
I think we should update to use `option.private_isArchived` directly. When we do migrate, this will be a one time change which should be okay but the code will be cleaner in the long run. 

My thinking is that the entire point of the report/option data is to have all the fields so that we don't have to get them using the reportID.

**Comment by srikarparsi** [comment]
Yup, this looks great. @situchan do you think you could help with another review and we can merge?

File: src/libs/OptionsListUtils.ts [hunk 7]
```diff
@@ -730,11 +728,11 @@ function createOption(
     let reportName;
     result.participantsList = personalDetailList;
     result.isOptimisticPersonalDetail = personalDetail?.isOptimisticPersonalDetail;
-    const reportNameValuePairs = ReportUtils.getReportNameValuePairs(report?.reportID);
     if (report) {
         result.isChatRoom = ReportUtils.isChatRoom(report);
         result.isDefaultRoom = ReportUtils.isDefaultRoom(report);
-        result.isArchivedRoom = ReportUtils.isArchivedRoom(report, reportNameValuePairs);
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        result.private_isArchived = report?.private_isArchived;
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:735]:
```suggestion
        result.private_isArchived = report.private_isArchived;
```

File: src/libs/OptionsListUtils.ts [hunk 8]
```diff
@@ -1588,7 +1586,7 @@ function orderOptions(options: ReportUtils.OptionData[], searchValue: string | u
                 if (preferChatroomsOverThreads && option.isThread) {
                     return 4;
                 }
-                if (!!option.isChatRoom || option.isArchivedRoom) {
+                if (!!option.isChatRoom || option?.private_isArchived) {
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:1589]:
```suggestion
                if (!!option.isChatRoom || option.private_isArchived) {
```

File: src/libs/OptionsListUtils.ts [hunk 9]
```diff
@@ -1833,7 +1831,7 @@ function getOptions(
     // - All archived reports should remain at the bottom
     const orderedReportOptions = lodashSortBy(filteredReportOptions, (option) => {
         const report = option.item;
-        if (option.isArchivedRoom) {
+        if (option?.private_isArchived) {
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:1834]:
```suggestion
        if (option.private_isArchived) {
```

File: src/libs/OptionsListUtils.ts [hunk 10]
```diff
@@ -1942,10 +1940,10 @@ function getOptions(
             }
 
             const isCurrentUserOwnedPolicyExpenseChatThatCouldShow =
-                reportOption.isPolicyExpenseChat && reportOption.ownerAccountID === currentUserAccountID && includeOwnedWorkspaceChats && !reportOption.isArchivedRoom;
+                reportOption.isPolicyExpenseChat && reportOption.ownerAccountID === currentUserAccountID && includeOwnedWorkspaceChats && !reportOption?.private_isArchived;
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:1943]:
```suggestion
                reportOption.isPolicyExpenseChat && reportOption.ownerAccountID === currentUserAccountID && includeOwnedWorkspaceChats && !reportOption.private_isArchived;
```

File: src/libs/OptionsListUtils.ts [hunk 11]
```diff
@@ -1942,10 +1940,10 @@ function getOptions(
...
 
             const shouldShowInvoiceRoom =
-                includeInvoiceRooms && ReportUtils.isInvoiceRoom(reportOption.item) && ReportUtils.isPolicyAdmin(reportOption.policyID ?? '', policies) && !reportOption.isArchivedRoom;
+                includeInvoiceRooms && ReportUtils.isInvoiceRoom(reportOption.item) && ReportUtils.isPolicyAdmin(reportOption.policyID ?? '', policies) && !reportOption?.private_isArchived;
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:1946]:
```suggestion
                includeInvoiceRooms && ReportUtils.isInvoiceRoom(reportOption.item) && ReportUtils.isPolicyAdmin(reportOption.policyID ?? '', policies) && !reportOption.private_isArchived;
```

File: src/libs/OptionsListUtils.ts [hunk 12]
```diff
@@ -2305,7 +2303,7 @@ function getHeaderMessageForNonUserList(hasSelectableOptions: boolean, searchVal
  * Helper method to check whether an option can show tooltip or not
  */
 function shouldOptionShowTooltip(option: ReportUtils.OptionData): boolean {
-    return (!option.isChatRoom || !!option.isThread) && !option.isArchivedRoom;
+    return (!option.isChatRoom || !!option.isThread) && !option?.private_isArchived;
```
Comment (situchan) [inline src/libs/OptionsListUtils.ts:2306]:
```suggestion
    return (!option.isChatRoom || !!option.isThread) && !option.private_isArchived;
```

File: src/libs/SidebarUtils.ts [hunk 13]
```diff
@@ -399,8 +400,7 @@ function getOptionData({
 
     const isThreadMessage =
         ReportUtils.isThread(report) && lastAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADD_COMMENT && lastAction?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
-
-    if ((result.isChatRoom || result.isPolicyExpenseChat || result.isThread || result.isTaskReport || isThreadMessage) && !result.isArchivedRoom) {
+    if ((result.isChatRoom || result.isPolicyExpenseChat || result.isThread || result.isTaskReport || isThreadMessage) && !result?.private_isArchived) {
```
Comment (situchan) [inline src/libs/SidebarUtils.ts:403]:
```suggestion
    if ((result.isChatRoom || result.isPolicyExpenseChat || result.isThread || result.isTaskReport || isThreadMessage) && !result.private_isArchived) {
```

**Review by situchan — COMMENTED** [review]
Looks good except redundant optional chainings

**Comment by srikarparsi** [comment]
Hi @cretadn22, can you please take a look at these comments when you get a chance and we can merge?

**Comment by blazejkustra** [comment]
Check [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1726169039084589) message on slack:
> The intention and expectation that generally you should fix these errors, even if you didn't introduce the usage.

**Comment by srikarparsi** [comment]
Thanks for bringing this up, I think it would be better to keep the scope of this PR small so that we can better regression test so I created a follow up [PR](https://github.com/Expensify/App/pull/49325) here to address the lint changes. I saw that we did the same thing [here](https://github.com/Expensify/App/pull/49316) so I'm going to go ahead and merge this.

**Review by srikarparsi — APPROVED** [review]
Thanks for the changes @cretadn22 and @situchan!

**Comment by srikarparsi** [comment]
Removing the emergency changes as the lint will be fixed in this follow up [PR](https://github.com/Expensify/App/pull/49325). More information in this [comment](https://github.com/Expensify/App/pull/46934#issuecomment-2354750686)

**Pull request merged**